test/manipulation.test.js

```diff
@@ -659,10 +659,60 @@ describe('manipulation', function() {
   });
 
   describe('updateOrCreate', function() {
+    var db = getSchema();
+    var Todo;
+
+    before(function setUp(done) {
+      Todo = db.define('Todo', {
+        content: String,
+      });
+      db.automigrate('Todo', done);
+    });
+    afterEach(function tearDown(done) {
+      db.automigrate(done);
+    });
+
     it('has an alias "patchOrCreate"', function() {
       StubUser.updateOrCreate.should.equal(StubUser.patchOrCreate);
     });
 
+    it('creates a model when one does not exist', function(done) {
+      Todo.updateOrCreate({ content: 'a' }, function(err, data) {
+        if (err) return done(err);
+
+        should.exist(data);
+        should.exist(data.content);
+        data.content.should.equal('a');
+
+        done();
+      });
+    });
+
+    it('updates a model if it exists', function(done) {
+      Todo.create({ content: 'a' }, function(err, todo) {
+        Todo.updateOrCreate({ id: 1, content: 'b' }, function(err, data) {
```

Comment:
`id` is not always `1`, different databases generate different id types (e.g. ObjectId in MongoDB). You should use `id: todo.id` instead.